libretroshare/src/rsitems/rsdiscovery2items.h

```diff
@@ -31,17 +31,24 @@
 
 #include "serialiser/rsserializer.h"
 
+// TODO: port to RsGossipDiscoveryItemType
 const uint8_t RS_PKT_SUBTYPE_DISC_PGP_LIST           = 0x01;
 const uint8_t RS_PKT_SUBTYPE_DISC_PGP_CERT           = 0x02;
 const uint8_t RS_PKT_SUBTYPE_DISC_CONTACT_deprecated = 0x03;
-const uint8_t RS_PKT_SUBTYPE_DISC_SERVICES           = 0x04;
 const uint8_t RS_PKT_SUBTYPE_DISC_CONTACT            = 0x05;
 const uint8_t RS_PKT_SUBTYPE_DISC_IDENTITY_LIST      = 0x06;
 
+enum class RsGossipDiscoveryItemType : uint8_t
```

Comment:
That is typically a case where mixing new style code with old style code is wrong: it makes it hard to see that these two IDs belogn to the list above. Please use the const uint8_t for them as well.

Reply:
well either we write well the new code and leave the conversion to enum for the other items for another PR, or we convert all the items to the enum, we already agreed that using all those integer types around is bad, because it make very difficult to understand what is passed around. Converting from enum to const uint8_t is walking back, and I don't do that...

Reply:
There is also one basic rule: when you modify a given file to add a single bit to it, try to respect the syntax, simply because it is more readable. For instance I've been using p3* for services without understanding why they are named this way because of consistency, same for serialization (before we rewrote *all* the serialization). Otherwise you're messing up the code more and more until it gets impossible to debug or even properly understand.